### DISCUSSION TITLE
[ 功能建议 ]账号申请工单

### DISCUSSION THREAD

**Discussion by dongqianzheng** [hhyo/Archery#1596] [General]
### 目前遇到的问题/使用障碍
平常建账号这种操作还是有点多的，如果总是手动操作的话，会有点麻烦。希望可以做成自动化，从申请-到审核-到创建-到邮件提醒。

### 希望如何解决/实现它
增加一个账号申请的工单。开发提申请，审核通过后，自动创建账号，账号创建好后账号密码邮件发送给申请人，密码信息自动记录到系统。
有点类似于 查询权限申请 和 实例管理-账号管理 的结合

**Comment by hhyo** [comment]
大家应该都有类似的管理需求

不过可以探讨一下是从数据库申请触发账号创建，还是说可以任意申请账号和授权

第二种能满足各种自定义场景，但如果是需要标准化的管理，个人偏向于前者，可以按照数据库维度生成对应的读、写、binlog账号，在建库的同时触发这个操作

**Comment by LeoQuote** [comment]
已经提供了ldap认证，普通企业应该就已经够了，申请全新账号建议要么走已经有的注册流程，要么通过外部程序实现，archery只提供入口

**Comment by dongqianzheng** [comment]
> 大家应该都有类似的管理需求
> 
> 不过可以探讨一下是从数据库申请触发账号创建，还是说可以任意申请账号和授权
> 
> 第二种能满足各种自定义场景，但如果是需要标准化的管理，个人偏向于前者，可以按照数据库维度生成对应的读、写、binlog账号，在建库的同时触发这个操作

在建库的时候就建对应的账号，确实是一个很好的思路。但是我们目前就有，一个账号会访问多个库的情况，当然这有可能是不规范或者设计问题。其次肯定也存在说，开发申请一个已经创建了很久的库，或者项目迁移，访问的ip地址变了(授权的ip访问范围比较小)。这种场景 在建库的时候触发建账号 就不满足了。可以考虑这两种结合起来。

**Comment by dongqianzheng** [comment]
> 已经提供了ldap认证，普通企业应该就已经够了，申请全新账号建议要么走已经有的注册流程，要么通过外部程序实现，archery只提供入口

~呃，你说的是审核平台的账号吧。我这里是说数据库的账号，主要是mysql。

**Comment by dongqianzheng** [comment]
这块的话，有计划做吗？

**Comment by dba3306** [comment]
我们公司MySQL资源申请是走workflow的，流程大概是开发提申请->leader审批->dba审批->创建资源，账号，库->加监控->加archery->结束

**Comment by dongqianzheng** [comment]
> 我们公司MySQL资源申请是走workflow的，流程大概是开发提申请->leader审批->dba审批->创建资源，账号，库->加监控->加archery->结束

workflow 是你们另外的一套系统吗？审批完成之后的创建等操作是自动完成还是手工操作的呢？

**Comment by dba3306** [comment]
> > 我们公司MySQL资源申请是走workflow的，流程大概是开发提申请->leader审批->dba审批->创建资源，账号，库->加监控->加archery->结束
> 
> workflow 是你们另外的一套系统吗？审批完成之后的创建等操作是自动完成还是手工操作的呢？

是另外一套开源的系统。审批完成后自动加入archery，这里做了点二次开发加了个接口。

**Comment by LeoQuote** [comment]
@dba3306 方便的话可以提个pr 把接口方面的代码提上来

**Comment by unknowissue** [comment]
> 是另外一套开源的系统。审批完成后自动加入archery，这里做了点二次开发加了个接口。

铁子，哪个开源系统呢，我看看咋搞
或者您直接提一下pr？
